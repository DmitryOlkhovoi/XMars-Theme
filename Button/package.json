{
  "name": "xmars-ui/Button",
  "private": true,
  "main": "../cjs/Button.js",
  "module": "../esm/Button.js",
  "types": "../types/Button.d.ts"
}
