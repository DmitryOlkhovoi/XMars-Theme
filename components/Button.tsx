import { omit } from 'underscore';
import React, { FunctionComponent, ButtonHTMLAttributes } from 'react';
import classnames from 'classnames';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
    primary?: boolean,
    additionalClass?: string,
}

const Button: FunctionComponent<Props> = (props) => {
    const {primary, additionalClass, children} = props;
    const classNames = classnames(
        'btn',
        {
            primary
        },
        additionalClass
    );

    return (
        <button className={classNames} {...omit(props, ['additionalClass', 'primary'])}>
            {children}
        </button>
    );
}

export default Button;