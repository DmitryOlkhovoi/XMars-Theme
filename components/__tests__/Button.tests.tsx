import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Button component', () => {
  test('Default render', () => {
    const component = renderer.create(
      <Button>
        Button
      </Button>,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Primary render', () => {
    const component = renderer.create(
      <Button primary>
        Button
      </Button>,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('additionalClass render', () => {
    const component = renderer.create(
      <Button additionalClass="foo bar">
        Button
      </Button>,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('onClick', () => {
    const onClick = jest.fn();

    const component = shallow(
      <Button onClick={onClick}>
        Button
      </Button>,
    );

    component.simulate('click');
  
    expect(onClick).toHaveBeenCalled();
  });
});