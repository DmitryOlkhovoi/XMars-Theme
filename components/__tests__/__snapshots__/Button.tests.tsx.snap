// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Default render 1`] = `
<button
  className="btn"
>
  Button
</button>
`;

exports[`Button component Primary render 1`] = `
<button
  className="btn primary"
>
  Button
</button>
`;

exports[`Button component additionalClass render 1`] = `
<button
  className="btn foo bar"
>
  Button
</button>
`;
